src/state/searchState.ts

```diff
@@ -92,20 +92,38 @@ export class SearchState {
 
       let lineIdx = lineToStartAt;
 
+      let numNewLines = (searchRE.match(/\\n/g) || []).length;
       // Start at the current line and wrap the document if we hit the end.
       outer:
       do {
-        const line = TextEditor.getLineAt(new Position(lineIdx, 0)).text;
+        let rangeStart = new Position(lineIdx, 0);
+        let rangeEnd = rangeStart.getLineEnd();
+        let lineLengths = [];
+        for (var i = 0; i < numNewLines; i++) {
+          lineLengths.push(rangeEnd.getLineEndIncludingEOL().character);
+          rangeEnd = rangeEnd.getDown(1).getLineEndIncludingEOL();
+        }
+        const line = TextEditor.getText(new vscode.Range(rangeStart, rangeEnd));
         let result = regex.exec(line);
 
         while (result) {
           if (this._matchRanges.length >= SearchState.MAX_SEARCH_RANGES) {
             break outer;
           }
+          let resultToPosition = function(resultIdx: number, lineIdx: number, lineLengths: Array<number>, result: RegExpExecArray){
```

Comment:
As a further style suggestion, it's good to prefer `=>` over `function` in the general case.